modules/test_infrastructure/main.tf

```diff
@@ -0,0 +1,152 @@
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_group
+resource "azurerm_resource_group" "this" {
+  count    = var.create_resource_group ? 1 : 0
+  name     = var.resource_group_name
+  location = var.location
+
+  tags = var.tags
+}
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/resource_group
+data "azurerm_resource_group" "this" {
+  count = var.create_resource_group ? 0 : 1
+  name  = var.resource_group_name
+}
+
+locals {
+  resource_group = var.create_resource_group ? azurerm_resource_group.this[0] : data.azurerm_resource_group.this[0]
+}
+
+# https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/blob/main/modules/vnet/README.md
+module "vnet" {
+  source = "../vnet"
+
+  for_each = var.vnets
+
+  name                   = each.value.name
+  create_virtual_network = each.value.create_virtual_network
+  resource_group_name    = local.resource_group.name
+  location               = var.location
+
+  address_space = each.value.address_space
+
+  create_subnets = each.value.create_subnets
+  subnets        = each.value.subnets
+
+  network_security_groups = each.value.network_security_groups
+  route_tables            = each.value.route_tables
+
+  tags = var.tags
+}
+
+# https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/blob/main/modules/vnet_peering/README.md
+module "vnet_peering" {
+  source   = "../vnet_peering"
+  for_each = { for k, v in var.vnets : k => v if v.hub_vnet_name != null }
+
+
+  local_peer_config = {
+    name                = "peer-${each.value.name}-to-${each.value.hub_vnet_name}"
+    resource_group_name = local.resource_group.name
+    vnet_name           = each.value.name
+  }
+  remote_peer_config = {
+    name                = "peer-${each.value.hub_vnet_name}-to-${each.value.name}"
+    resource_group_name = try(each.value.hub_resource_group_name, local.resource_group.name)
+    vnet_name           = each.value.hub_vnet_name
+  }
+
+  depends_on = [module.vnet]
+}
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/network_interface
+resource "azurerm_network_interface" "vm" {
+  for_each = var.spoke_vms
+
+  name                = each.value.interface_name
+  location            = var.location
+  resource_group_name = local.resource_group.name
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = module.vnet[each.value.vnet_key].subnet_ids[each.value.subnet_key]
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+locals {
+  password = sensitive(var.authentication.password)
+}
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine
+resource "azurerm_linux_virtual_machine" "this" {
```

Comment:
Hmm, since it obviously makes sense to provide redundancy options for VM-Series, I'm not sure if we need that for VMs required only for testing purposes. Let's discuss it.